### ISSUE TITLE
FNX3-16309 ⁃ [Bug] crash on deleting history

### ISSUE THREAD

**Issue by cadeyrn** [mozilla-mobile/fenix#2419]
### Steps to reproduce
I wanted to delete my history and pressed the button in the library to delete the full history. At first nothing happened (maybe it took a long time because of a very large history? The history was deleted when I opened the library later) so I impatiently clicked the button several times. Then Fenix crashed:

```
2019-05-10 21:26:28.782 10989-11144/org.mozilla.fenix.debug E/AndroidRuntime: FATAL EXCEPTION: DefaultDispatcher-worker-1
    Process: org.mozilla.fenix.debug, PID: 10989
    java.lang.IllegalStateException: Fragment HistoryFragment{5113559 (06560a8f-a923-4e6c-8533-d3e14281c03e)} not attached to a context.
        at androidx.fragment.app.Fragment.requireContext(Fragment.java:760)
        at org.mozilla.fenix.ext.FragmentKt.getRequireComponents(Fragment.kt:13)
        at org.mozilla.fenix.library.history.HistoryFragment.reloadData(HistoryFragment.kt:209)
        at org.mozilla.fenix.library.history.HistoryFragment$onStart$1$1.invokeSuspend(HistoryFragment.kt:132)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:238)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:594)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:60)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:742)
```

### Device information
* Android device: HTC U11 / Android 8.0
* Fenix version: Fenix revision e1cdeffe8c7168141a586b7c9126555df4c0fb53

**Comment by ekager** [comment]
Sounds like we could cancel out the reloadData jobs if another job is added or if user navigates away

**Comment by softvision-miralobontiu** [comment]
Verified as fixed on the latest build 1.0.1922 (#11500615), with Samsung Galaxy S8 (Android 9), and Nokia 6 (Android 7.1.1).
